feat: add sftpsend plugin

matthieumarrast opened the pull request.

thefab commented on plugins/sftpsend/main.py:
```diff
@@ -0,0 +1,160 @@
+            raise Exception("you have to set a valid sftp_username value")
+        if self.sftp_password == "FIXME" or self.sftp_password == "null" \
+                or self.sftp_password == "":
+            raise Exception("you have to set a valid sftp_password value")
```
I don't know SFTP very much but if you use a key_file, you don't need password, do you?

matthieumarrast replied:
OK fixed

thefab reviewed (changes requested):
we also need the requirements3.txt files

as we don't want any surprise if we rebuild the plugin two years later in a production branch if you see what I mean ;-)

thebaptiste commented:
Two remarks : 
- for integration, master and release_1.1 branches, we probably don't need to force cryptography < 3 (for these branches, the cryptography version in mfext is 3.3.2). I think we probably don't need to have cryptography at all in requirement files because it's provided by mfext
- pysftp is a very old project which hasn't been released since July 2016, it seems that it has not been tested with python > 3.4 and it is probably not maintained anymore. May be a more recent sftp project would be better ? There are several "sftp" projects in pypi, such as sftp-py. But it's just a remark, may be pysftp is great.

thefab commented:
yes @thebaptiste!

@matthieumarrast is going to push the virtualenv deletion
(it was here only because it's a port of a plugin developed for 0.9)

matthieumarrast commented:
> 
> 
> yes @thebaptiste!
> 
> @matthieumarrast is going to push the virtualenv deletion
> (it was here only because it's a port of a plugin developed for 0.9)

DONE

matthieumarrast commented:
> 
> 
> Two remarks :
> 
>     * for integration, master and release_1.1 branches, we probably don't need to force cryptography < 3 (for these branches, the cryptography version in mfext is 3.3.2). I think we probably don't need to have cryptography at all in requirement files because it's provided by mfext ( < 3 for release_1.0, > 3 for release >= 1.1)
> 
>     * pysftp is a very old project which hasn't been released since July 2016, it seems that it has not been tested with python > 3.4 and it is probably not maintained anymore. May be a more recent sftp project would be better ? There are several "sftp" projects in pypi, such as sftp-py. But it's just a remark, may be pysftp is great and there will probably never be a new version of pysftp, even in two years, so we don't risk any surprise !

Indeed I have removed `pysftp` and `cryptography`, replaced by `paramiko`.
